pep517/build.py

```diff
@@ -3,25 +3,50 @@
 import argparse
 import logging
 import os
-import contextlib
 import pytoml
 import shutil
-import errno
-import tempfile
 
 from .envbuild import BuildEnvironment
 from .wrappers import Pep517HookCaller
+from .dirtools import tempdir, mkdir_p
 
 log = logging.getLogger(__name__)
 
 
-@contextlib.contextmanager
-def tempdir():
-    td = tempfile.mkdtemp()
+def validate_system(system):
+    """
+    Ensure build system has the requisite fields.
+    """
+    required = {'requires', 'build-backend'}
```

Comment:
seems overzealous, can't we fallback for the build-backend key to setuptools legacy as pip does?

Reply:
I would say even for requires we should just inject ``setuptools >= 41.0.0`` and ``wheel >= 0.30.0``.

Reply:
This function is simply about ensuring the requisite fields are there. There is another function, `compat_system`, that supplies the fallback. If we could say that "there is no case where the fallback would not be desirable", then I agree, the fallback should be derived unconditionally and there's no need to validate. But since the existing behavior is to fail when an invalid backend is supplied, I wanted to retain that behavior for existing uses and for the use-case herein, support the fallback.

Reply:
I've filed #49 to follow up on this consideration separately.